typegate/src/planner/args.ts

```diff
@@ -832,29 +863,63 @@ class JsonSchemaBuilder {
   }
 
   build(typeNode: TypeNode): JSONSchema {
+    let resultSchema: JSONSchema = {};
```

Comment:
Is there a need to have this temporary variable instead of the return in all the cases?

Reply:
Yes, it helps handling `case Type.OPTIONAL` and `case Type.OBJECT` as I mutate the properties before a return. Possibly reducible to return statements but that would be more of an aesthetic issue I think as I'd have to create more local variables.